src/deepsparse/schedulers/continuous_batching_scheduler.py

```diff
@@ -167,16 +167,30 @@ def add_engine_operator(
                 continue  # already added
 
             override_model_path = None
-            # text generation/NLEngineOperator specific; could add generic method
-            # for all engine_operators, if desired
-            if hasattr(engine_operator, "override_model_inputs"):
-                override_model_path = engine_operator.override_model_inputs(
-                    model_path=engine_operator.model_path, batch_size=batch_size
+            kwargs = {}
+
+            if engine_operator.__class__.__name__ == "NLEngineOperator":
+                (
+                    override_model_path,
+                    additional_outputs,
+                ) = engine_operator.override_model_inputs(
+                    model_path=engine_operator.model_path,
+                    batch_size=batch_size,
+                    return_additional_outputs=True,
                 )
 
-            # will break for internal kv_cache; needs additional argument
+                if (
+                    engine_operator._engine_type == DEEPSPARSE_ENGINE
+                    and engine_operator.internal_kv_cache
+                ):
+                    output_indices_to_be_cached = additional_outputs.get(
+                        "output_indices_to_be_cached"
+                    )
+                    if any(output_indices_to_be_cached):
```

Comment:
in which scenario `any(output_indices_to_be_cached)==False` given that
` (engine_operator._engine_type == DEEPSPARSE_ENGINE and engine_operator.internal_kv_cache)== True`
?

Reply:
this condition is from v1. if they're not mutually exclusive, I can combine